diff --git a/tmpl/admin2025/cms/edit_template.tmpl b/tmpl/admin2025/cms/edit_template.tmpl
index 43690862a..d7439a933 100644
--- a/tmpl/admin2025/cms/edit_template.tmpl
+++ b/tmpl/admin2025/cms/edit_template.tmpl
@@ -361,7 +361,11 @@
 
 <mt:setvarblock name="html_head" append="1">
 <mtapp:script path="codemirror/lib/codemirror.js">
+<mtapp:script path="codemirror/addon/search/searchcursor.js">
+<mtapp:script path="codemirror/addon/search/search.js">
+<mtapp:script path="codemirror/addon/dialog/dialog.js">
 <mtapp:stylesheet path="codemirror/lib/codemirror.css">
+<mtapp:stylesheet path="codemirror/addon/dialog/dialog.css">
 
 <mtapp:script path="codemirror/mt/xml.js">
 <mtapp:script path="codemirror/mt/javascript.js">
